table/block_based/block_based_table_iterator.cc

```diff
@@ -74,7 +79,14 @@ void BlockBasedTableIterator::SeekImpl(const Slice* target) {
   } else {
     // Need to use the data block.
     if (!same_block) {
-      InitDataBlock();
+      if (async_call) {
+        AsyncInitDataBlock(/* first_pass= */ true);
+        async_status_ = Status::TryAgain();
```

Comment:
`block_iter_.status().IsIncomplete()` check is there in the code so I used `TryAgain` to avoid the status conflicts.